### PR TITLE
Allen/etimated score widget TFL-1029

### PR DESCRIPTION
**Pull request by Allensy**

### CODE REVIEW

File: src/scss/imports/_colors.scss [hunk 1]
```diff
@@ -76,6 +83,10 @@ $fontColor11: #968E8E;
 $fontColor12: #898989;
 $fontColor13: #7d7d7d;
 $fontColor14: #acacac;
+$fontColor15: #968E8E;
+$fontColor16: #898989;
```
Comment (igormas) [inline src/scss/imports/_colors.scss]: Already exists

File: src/components/estimatedScoreWidget/svg/goals-top-icon.svg [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<svg class="goals-top-icon-wrapper" x="0px" y="0px" viewBox="-632.7 361.9 200 200">
+    <style type="text/css">
+        .goals-top-icon-wrapper .st0{fill:none;}
```
Comment (igormas) [inline src/components/estimatedScoreWidget/svg/goals-top-icon.svg]: add init width, change the class name to estimated-score-widget-goals-icon

File: src/components/estimatedScoreWidget/svg/close-popup.svg [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<svg
+    x="0px"
+    y="0px"
+    viewBox="-596.6 492.3 133.2 133.5">
+<path class="st0"/>
```
Comment (igormas) [inline src/components/estimatedScoreWidget/svg/close-popup.svg]: add style tag

File: src/components/estimatedScoreWidget/module.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        'znk.infra.estimatedScore',
+        'znk.infra.scoring',
+        'znk.infra.svgIcon',
+        'znk.infra-web-app.userGoals',
+        'znk.infra-web-app.userGoalsSelection'
+    ]).config([
+        'SvgIconSrvProvider',
+        function (SvgIconSrvProvider) {
+            var svgMap = {
+                'goals-top-icon': 'components/estimatedScoreWidget/svg/goals-top-icon.svg',
```
Comment (igormas) [inline src/components/estimatedScoreWidget/module.js]: add estimated-score-widget prefix to svg names

File: src/components/estimatedScoreWidget/main.scss [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+                        border-top-right-radius: 10px;
+                    }
+                    &.user-goal-met {
+                        z-index: 2;
+                        background: transparent;
+                        border-right: {
+                            width: 2px;
+                            style: solid;
+                        }
+                        //&.math {
```
Comment (igormas) [inline src/components/estimatedScoreWidget/main.scss]: delete

File: src/components/estimatedScoreWidget/main.scss [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+                border-bottom: {
+                    style: solid;
+                    color: $borderColor15;
+                    width: 2px;
+                }
+                border-left: {
+                    style: solid;
+                    width: 2px;
+                }
+                //&.math {
```
Comment (igormas) [inline src/components/estimatedScoreWidget/main.scss]: delete

File: src/components/estimatedScoreWidget/locale/en.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        "COMPOSITE_SCORE": "Total Score:",
+        "GOAL_SCORE": "Goal Score:",
+        "EDIT_MY_GOALS": "Edit my goals",
+        "SUBJECT_0": "Math",
+        "SUBJECT_1": "English",
+        "SUBJECT_7": "Verbal",
+        "SUBJECT_8": "Essay",
+        "UNFINISHED_DIAGNOSTIC_TITLE": "Set your initial score by completing the diagnostic test."
+    },
+    "SETTING.EDIT_GOALS": {
```
Comment (igormas) [inline src/components/estimatedScoreWidget/locale/en.json]: why there is a dot ?

File: src/components/estimatedScoreWidget/locale/en.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{
+    "ESTIMATED_SCORE_WIDGET_DIRECTIVE": {
+        "TITLE": "SAT Estimated Score",
+        "PTS_TO_GO": "{{pts}} pts to go",
+        "YOUR_GOAL": "Your Goal: {{goal}}",
+        "GOAL_REACHED": "you've reached your goal!",
+        "COMPOSITE_SCORE": "Total Score:",
+        "GOAL_SCORE": "Goal Score:",
+        "EDIT_MY_GOALS": "Edit my goals",
+        "SUBJECT_0": "Math",
```
Comment (igormas) [inline src/components/estimatedScoreWidget/locale/en.json]: the subjects name already defined int the app:
    "SUBJECTS":{
        "0": "math",
        "7": "verbal",
        "8": "essay"
    },

File: src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                },
+                link: function (scope, element, attrs, ngModelCtrl) {
+                    $translatePartialLoader.addPart('estimatedScoreWidget');
+                    scope.d = {};
+
+                    var isNavMenuFlag = (scope.isNavMenu === 'true');
+
+                    var getLatestEstimatedScoreProm = EstimatedScoreSrv.getLatestEstimatedScore();
+                    var getSubjectOrderProm = EstimatedScoreWidgetSrv.getSubjectOrder();
+                    // var getEstimatedScoreCompositeProm = EstimatedScoreSrv.getCompositeScore();
```
Comment (igormas) [inline src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js]: delete

File: src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+                            getSubjectOrderProm
+
+                        ]).then(function (res) {
+                            var estimatedScore = res[0];
+                            var isDiagnosticCompleted = res[1];
+                            var subjectOrder = res[2];
+
+                            scope.d.isDiagnosticComplete = isDiagnosticCompleted;
+
+                            // scope.d.estimatedCompositeScore = isDiagnosticCompleted ? res[1].compositeScoreResults || '-' : '-';todo need to figure out what it do
```
Comment (igormas) [inline src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js]: delete if not needed

File: src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+                    // var getEstimatedScoreCompositeProm = EstimatedScoreSrv.getCompositeScore();
+                    // var isDiagnosticCompletedProm = WorkoutsDiagnosticFlow.isDiagnosticCompleted();todo implement once diagnostic service will be ready
+                    var subjectEnumToValMap = SubjectEnum.getEnumMap();
+
+                    if (isNavMenuFlag) angular.element.addClass(element[0], 'is-nav-menu');
+
+                    function adjustWidgetData(userGoals) {
+                        $q.all([
+                            getLatestEstimatedScoreProm,
+                            $q.when(false),
```
Comment (igormas) [inline src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js]: ?

File: src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+                    }
+
+                    function calcPercentage(correct) {
+                        var scoringLimits = UserGoalsService.getScoringLimits();
+                        var maxEstimatedScore = typeof scoringLimits.subjects[Object.getOwnPropertyNames(scoringLimits.subjects)] !== 'undefined' ? scoringLimits.subjects[Object.getOwnPropertyNames(scoringLimits.subjects)].max : scoringLimits.subjects.max;
+                        return (correct / maxEstimatedScore) * 100;
+                    }
+
+                    // TODO: this should come from a service, duplicated from znk-header
+                    scope.d.showGoalsEdit = function () {
```
Comment (igormas) [inline src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js]: ?

File: src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+                                    }, 1200);
+                                });
+                            }
+
+                            previousValues = scope.d.widgetItems;
+                        });
+                    }
+
+                    function calcPercentage(correct) {
+                        var scoringLimits = UserGoalsService.getScoringLimits();
```
Comment (igormas) [inline src/components/estimatedScoreWidget/directives/estimatedScoreWidget.directive.js]: Isnt it async ?

**Comment by assafshp** [comment]
jenkins-test

**Pull request merged**